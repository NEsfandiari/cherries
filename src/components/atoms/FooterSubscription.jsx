import React, { Component } from 'react'
import Styled from 'styled-components'

const Container = Styled.div`
  display: flex;
  flex-basis: 50%;
  max-height: 200px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  button {
    
  }
  input{
    width: 16rem;
  }
`

class FooterSubscription extends Component {
  state = {
    email: '',
    status: 'Sign Up',
  }

  handleSubmit = e => {
    e.preventDefault()
    let email = this.state.email

    // Netlify Form Encoding for Email Subscriberes
    const encode = data => {
      return Object.keys(data)
        .map(
          key => encodeURIComponent(key) + '=' + encodeURIComponent(data[key])
        )
        .join('&')
    }

    fetch('/', {
      method: 'POST',
      headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
      body: encode({ 'form-name': 'email', email }),
    })
    this.setState({ email: '', status: 'Added!' })
  }

  handleChange = e => {
    this.setState({
      email: e.target.value,
    })
  }

  render() {
    return (
      <Container>
        <div>
          <h4>{this.props.title}</h4>
          <p style={{ maxWidth: '20.8rem' }}>{this.props.text}</p>
          <form onSubmit={this.handleSubmit} name="email" netlify>
            <label htmlFor="email">Email</label>
            <br />
            <input
              type="email"
              onChange={this.handleChange}
              value={this.state.email}
            />
            <button
              style={{
                marginLeft: '12px',
                height: '32px',
                backgroundColor: '#13CE66',
                color: '#47525E',
                border: 'none',
                width: '6.5rem',
                paddingTop: '2px',
              }}
            >
              {this.state.status}
            </button>
          </form>
        </div>
      </Container>
    )
  }
}

export default FooterSubscription
