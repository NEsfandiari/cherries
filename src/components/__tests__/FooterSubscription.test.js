import React from 'react'
import renderer from 'react-test-renderer'
import FooterSubscription from '../atoms/FooterSubscription'

describe('FooterSubscription', () =>
  it('renders correctly', () => {
    const text = 'Footer'
    const title = 'Enter Footer text here, boiiiiii'

    const tree = renderer
      .create(<FooterSubscription title={title} text={text} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))
