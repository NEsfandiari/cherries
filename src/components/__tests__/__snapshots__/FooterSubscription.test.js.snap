// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterSubscription renders correctly 1`] = `
<div
  className="sc-bdVaJa cncvlg"
>
  <div>
    <h4>
      Enter Footer text here, boiiiiii
    </h4>
    <p
      style={
        Object {
          "maxWidth": "20.8rem",
        }
      }
    >
      Footer
    </p>
    <form>
      <label
        htmlFor="email"
      >
        Email
      </label>
      <br />
      <input
        style={
          Object {
            "width": "16rem",
          }
        }
        type="email"
      />
      <button
        style={
          Object {
            "backgroundColor": "#13CE66",
            "border": "none",
            "color": "#47525E",
            "height": "32px",
            "marginLeft": "12px",
            "paddingTop": "2px",
            "width": "6.5rem",
          }
        }
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
